### ISSUE TITLE
[Accessibility Audit] Code Intel: Finding implementations for a definition

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#33511]
### Steps to replicate journey

Taken from the [user journey list document](https://docs.google.com/document/d/1kA6aVOAgID_uPm-d6uEC1DG6WHjJ8fPT0UAXQcML_KQ/edit?usp=sharing). Use this for further context.

- Hover over an identifier.
- Click on the Find implementations button in the Code Intel Popover.
- Optional: Potentially increase height of the Implementations panel.
- Select repository and file of interest from the Implementations panel.
- Click on code snippet to navigate to reference.

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by sourcegraph-bot-2** [comment]
Heads up @macraig - the "team/code-intelligence" label was applied to this issue.

**Comment by Numbers88s** [comment]
- takes long to reach document
- because of the tab hierarchy is faster to navigate backwards to get to pop up.
- we found an lsif go bug
- focus seems off after clicking on find implementation button. It is not in reference panel
- after selecting close it says that its still in tab panel which is misleading because it was still in the close button.
- In the button group, it mentions close button twice. Once for the image and the second one for the text
- tabbing thru heading in the reference panel seems to work great.
- If I clicked on show implementation button then we should show implementation results first instead of references.
- when tab from implementation title it seems to go to code search logo it should go to the implementation list
- you can get to heading but you cannot go into the nested buttons under the heading with control+option+right
    - heading seems to be stuck in an island. Cannot go back to the level to the close button or list
- Maybe we need to have a prefix when you go thru the labels to know if its a repository/directory/file
- Ctrl+option+down doesn’t seem to work on a list within the reference panel
- maybe there should be a header 3 to Searcher in reference panel to navigate to it quickly.
- says 1 displayed when filtered but its potentially more useful is to provide more context like how many more are there.
- Instead of saying button it should say clickable. I wonder if we are using buttons out of convenience but we should be using links instead.
- When clicking on the implementation file it should say opening or callout the action that the user is taking. After the action is done the focus should be on the file the user just opened.
- No feedback for user events. For example:
    - we accidentally closed the code view but we had no way of knowing what happened unless we see it.
    - when we try to open a file in code view there is no feedback that we opened a code view
    - When we change the code view there is no feedback that we changed the code view. Focus didn’t change to the selected view. So now we have to scroll to the end to get to the previously selected code view.
    - we should announce what table we have selected, there is no feedback.
- Hierarchy seems off once you are in code view. If we go back we expect to go to the reference side panel instead to the top of the file.

Restricted view
- its pretty unreadable as it is very small. We need to hide it to use it.
- There is no horizontal scroll button.
- We cannot see find reference button
- Positioning of elements seems broken. Everything is too close together.

**Comment by varungandhi-src** [comment]
Closing this; most of the issues uncovered here were related to the reference panel. For consistency, those are linked from the "Finding references for a definition" user journey.

I've filed issues for everything we found.
